collect_app/src/main/java/org/odk/collect/android/tasks/ServerPollingWorker.java

```diff
@@ -200,10 +211,17 @@ private boolean isDeviceOnline() {
     public static void pollServerIfNeeded() {
```

Comment:
@grzesiek2010 Hey! Why is `pollServerIfNeeded` necessary in the context of this job? I was thinking to remove it since the `NetworkReceiver` is being removed. However, the `AutoSendWorker` is now the new `NetworkReceiver` so if there's a solid rationale behind this it could be tied back in.

Reply:
Let's imagine a case that you have selected `Every 24 hours` option to check for updates. You did that let's say on Monday at 1pm so you can expect next checks on Tuesday, Wednesday etc at the same time. But if your device has no internet connection at this particular time you lost your chance and you have to wait another 24 hours. The method you asked about is used to run the job immediately after connecting to the internet if the last attempt failed.